### ISSUE TITLE
File attached to entity ends forbidden to see after session expired

### ISSUE THREAD

**Issue by desirelabs** [directus/directus#11138]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

In directus 9 admin panel,
given an entity with a file upload field, uploading and viewing the file works.
After the session expires and refreshing it, viewing the same image both in file nor in the entity becomes forbidden.

### To Reproduce

- create an entity with a file upload field
- create a content based on this entity
- upload a file via the input
- go away to wait for the session to expire
- refresh the page

### Errors Shown

![image](https://user-images.githubusercontent.com/24223583/149977011-024bb6eb-d497-4c10-ad1e-3b3b5d698bad.png)

### What version of Directus are you using?

9.4.3

### What version of Node.js are you using?

14.x

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### What operating system are you using?

MacOS

### How are you deploying Directus?

heroku

**Comment by joselcvarela** [comment]
Hey
I tried on my side and was not able to reproduce this.

Every time I hit reload button, a POST /auth/refresh is made and I can see the file properly.
Once the refresh_token expires it logs me out of the app.

Are your user an Administrator or has another role?

**Comment by desirelabs** [comment]
Hi, my user is Administrator. And I set reading permissions on the entity and directus_files public.

**Comment by desirelabs** [comment]
I uploaded a capture of my current public permissions.

**Comment by joselcvarela** [comment]
Can you post your `ACCESS_TOKEN_TTL` and `REFRESH_TOKEN_TTL` configurations?

**Comment by joselcvarela** [comment]
I just found out this happens when file is missing on folder configured by `STORAGE_LOCAL_ROOT`. By default, this folder is `uploads`.
So for example, I have an asset with `78692c1f-1d8e-4cc1-aee8-a26979ac9b42` ID and if this name does not appear inside `uploads` folder, I have what you mentioned. 

@rijkvanzanten I found that if a record or file is missing the status code is always 403.
But, shouldn't it be 404 Not Found if the user has permissions to it?
